src/pysonar_scanner/configuration/toml_file.py

```diff
@@ -0,0 +1,58 @@
+#
+# Sonar Scanner Python
+# Copyright (C) 2011-2024 SonarSource SA.
+# mailto:info AT sonarsource DOT com
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+#
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+from pathlib import Path
+from typing import Dict
+import os
+import tomli
+
+from pysonar_scanner.configuration import properties
+
+
+def load(base_dir: Path) -> Dict[str, str]:
+    filepath = base_dir / "pyproject.toml"
+    if not os.path.isfile(filepath):
+        return {}
+
+    try:
+        with open(filepath, "rb") as f:
+            toml_dict = tomli.load(f)
+
+        # Look for configuration in the tool.sonar section
+        if "tool" in toml_dict and "sonar" in toml_dict["tool"]:
+            # Flatten nested dictionaries into dot notation keys
+            sonar_config = toml_dict["tool"]["sonar"]
+            result = {}
+            python_to_java_names = {prop.python_name(): prop.name for prop in properties.PROPERTIES}
+
+            def flatten_dict(config, prefix=""):
+                for key, value in config.items():
+                    if isinstance(value, dict):
+                        flatten_dict(value, f"{prefix}{key}.")
+                    else:
+                        property_name = f"sonar.{prefix}{key}"
+                        property_name = python_to_java_names.get(property_name, property_name)
+                        result[property_name] = value
+
+            flatten_dict(sonar_config)
+            return result
+        return {}
+    except Exception:
+        # If there's any error parsing the TOML file, return empty dict
+        return {}
```

Comment:
nitpick: I would potentially make sense to have a debug logging statement here. But this could also be done as part of [SCANPY-135](https://sonarsource.atlassian.net/browse/SCANPY-135)
```suggestion
        logging.debug(...)
        return {}
```

Reply:
I agree! I'll refer to that ticket, I think it makes sense to add logging in a consistent way and SCANPY-135 is most likely the way to do that.